sdl2-vga-terminal/test/snapshotTest.cpp

```diff
@@ -27,33 +30,19 @@ SDL_Surface* getScreenshot(const VgaTerminal& term)
 	return snapshot;
 }
 
-TEST(VgaTerminal, Snapshot)
+void snapShotTest(VgaTerminal& term, const std::string& snapshotFilename)
 {
-	SDL_Init(SDL_INIT_VIDEO);
-	IMG_Init(IMG_INIT_PNG);
-	std::string termTitle = "Hello Test";
-	VgaTerminal term = VgaTerminal(termTitle, 0, -1, 0);
-	term.showCursor = false;
-	std::string snapshotFilename = generateSnapshotFilename();
-
-	for (int i = 0; i < 256; i++) {
-		term.write(static_cast<char>(i), i, 255 - i);
-	}
-
-	term.writeXY(32, 11, "���������������ͻ", 14, 1);
-	term.writeXY(32, 12, "� Hello World!! �", 14, 1);
-	term.writeXY(32, 13, "���������������ͼ", 14, 1);
 	term.render();
-	SDL_Delay(1000);
+
 	SDL_Surface* snapshot = getScreenshot(term);
 
 #ifdef TEST_DUMP_SNAPSHOT
 	GTEST_LOG_(INFO) << "Dumping snapshot: " << snapshotFilename;
 	IMG_SavePNG(snapshot, ("snapshot/" + snapshotFilename).c_str());
-	SDL_FreeSurface(snapshot);
 #else
 	SDL_Surface* image = IMG_Load(("snapshot/" + snapshotFilename).c_str());
-
+	ASSERT_FALSE(NULL == image);
+	ASSERT_FALSE(NULL == image->format);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Either the condition 'NULL==image' is redundant or there is possible null pointer dereference: image.](https://app.codacy.com/manual/Raffaello/sdl2-vga-terminal/pullRequest?prid=5427759)